internal/provider/data_source_record.go

```diff
@@ -0,0 +1,80 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCloudflareRecord() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceCloudflareRecordRead,
+		Schema: map[string]*schema.Schema{
+			"zone_id": {
+				Description: "The zone identifier to target for the resource.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"record_id": {
+				Description: "The record identifier to target for the resource.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"hostname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"proxied": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"ttl": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"proxiable": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"locked": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"zone_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceCloudflareRecordRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
i don't quite think this is usable. if someone already has the ID, wouldn't they use it? usually, data sources are used for when you have a fuzzy idea of what you're after but can't pin point it. i.e. a DNS record for "@" in a particular zone.